### PR TITLE
Request DB reset when dev/test/sim network restart is detected

### PR DESCRIPTION
**Pull request by coderofstuff**
This detects if the current Genesis (as defined in code) is in the local active consensus DB - which it always should be. If it isn't, then the network has restarted and the local DB must be deleted. Only applied to non-mainnet networks as mainnet is never retarted.

### CODE REVIEW

**Review by someone235 — CHANGES_REQUESTED** [review]

File: kaspad/src/daemon.rs [hunk 1]
```diff
@@ -243,6 +244,54 @@ do you confirm? (answer y/n or pass --yes to the Kaspad command line to confirm
...
+
+        match active_consensus_dir_name {
+            Some(dir_name) => {
+                let consensus_db = kaspa_database::prelude::ConnBuilder::default()
+                    .with_db_path(consensus_db_dir.clone().join(dir_name))
+                    .with_files_limit(1)
+                    .build()
+                    .unwrap();
+
+                if consensus_db
```
[the remark this answers is not in the document]
Comment (coderofstuff) [reply kaspad/src/daemon.rs:261]: @someone235 do you mean something more like this?

```
                let consensus_db = kaspa_database::prelude::ConnBuilder::default()
                    .with_db_path(consensus_db_dir.clone().join(dir_name))
                    .with_files_limit(1)
                    .build()
                    .unwrap();

                let consensus_storage = ConsensusStorage::new(consensus_db, config.clone());

                if consensus_storage.headers_store.has(config.genesis.hash) {
                    trace!("Genesis is found in active consensus DB. No action needed.");
                }
```

[the remark this answers is not in the document]
Comment (someone235) [reply kaspad/src/daemon.rs:261]: I thought about using `DbHeadersStore::new`

[the remark this answers is not in the document]
Comment (coderofstuff) [reply kaspad/src/daemon.rs:261]: Ahh, right, I can actually just use that directly. Thanks:

```
                let consensus_db = kaspa_database::prelude::ConnBuilder::default()
                    .with_db_path(consensus_db_dir.clone().join(dir_name))
                    .with_files_limit(1)
                    .build()
                    .unwrap();

                let headers_store = DbHeadersStore::new(consensus_db, 0);

                if headers_store.has(config.genesis.hash) {
                    trace!("Genesis is found in active consensus DB. No action needed.");
                }
```

File: kaspad/src/daemon.rs [hunk 2]
```diff
@@ -243,6 +244,54 @@ do you confirm? (answer y/n or pass --yes to the Kaspad command line to confirm
         .build()
         .unwrap();
 
+    if args.testnet && args.testnet_suffix == 11 {
```
[the remark this answers is not in the document]
Comment (coderofstuff) [reply kaspad/src/daemon.rs:252]: `args.mainnet` doesn't seem to be in args. Maybe `if args.testnet || args.simnet || args.devnet`?

**Review by someone235 — APPROVED** [review]

**Pull request merged**